### PR TITLE
fix: select treeselect value validation

### LINKED ISSUE #10757: [Bug]: Select widget treating value `0` as false and hence causing validation issues in the form
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I am passing the following array as options to the widget. This widget's REQUIRED is set to true w.r.t the form widget it is inside in.

```
[{
  label: 'Active',
  value: 1
}, {
  label: 'Inactive',
  value: 0
}]

```

I also have a form button which has "Disabled on Invalid Forms" set to true.
Now when I select the `Inactive` option from the select widget, the "Submit" button is seen as disabled. This issue wasn't there earlier.

### Steps To Reproduce

1. add a select widget in the form widget with the above data for options. 
2. set the select field to required
3. set the form to `Disabled on Invalid Forms`
4. select the `invalid` option with value 0
5. observe that the form button is still disabled

### Environment

Production

### Version

Self-Hosted

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

- Fix Validation for Select and Treeselect WIdget having a value of `0`

Fixes #10757

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/select-treeselect-value-validation 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.01 **(0)** | 36.49 **(-0.01)** | 34.73 **(0)** | 55.41 **(0)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/widgets/DropdownWidget/widget/index.tsx [hunk 1]
```diff
@@ -278,7 +278,7 @@ class DropdownWidget extends BaseWidget<DropdownWidgetProps, WidgetState> {
 
   static getDerivedPropertiesMap() {
     return {
-      isValid: `{{this.isRequired  ? !!this.selectedOptionValue : true}}`,
+      isValid: `{{this.isRequired  ? !!this.selectedOptionValue || this.selectedOptionValue === 0 : true}}`,
```
Comment (sbalaji1192) [inline app/client/src/widgets/DropdownWidget/widget/index.tsx:281]: `{{(()=>{return !this.isRequired || !_.isNil(this.selectedOptionValue)})()}}`

File: app/client/src/widgets/SingleSelectTreeWidget/widget/index.tsx [hunk 2]
```diff
@@ -302,8 +302,8 @@ class SingleSelectTreeWidget extends BaseWidget<
     return {
       selectedOptionLabel: `{{  this.selectedLabel[0] }}`,
       selectedOptionValue:
-        '{{  JSON.stringify(this.options).match(new RegExp(`"value":"${this.selectedOption}"`), "g") ? this.selectedOption : undefined  }}',
-      isValid: `{{this.isRequired  ? !!this.selectedOptionValue?.length : true}}`,
+        '{{  JSON.stringify(this.options).match(new RegExp(`"value":${Number.isFinite(this.selectedOption) ? this.selectedOption : `"${this.selectedOption}"` }`), "g") ? this.selectedOption : undefined  }}',
+      isValid: `{{this.isRequired  ? !!this.selectedOptionValue || this.selectedOptionValue === 0 : true}}`,
```
Comment (sbalaji1192) [inline app/client/src/widgets/SingleSelectTreeWidget/widget/index.tsx:306]: `{{(()=>{return !this.isRequired || !_.isNil(this.selectedOptionValue)})()}}`

File: app/client/src/widgets/DropdownWidget/widget/index.tsx (hunk 1 above)
Comment (Tooluloope) [reply app/client/src/widgets/DropdownWidget/widget/index.tsx:281]:
```
{
label: "label",
value: ""
}
```

would return `true`.
I dont think we should allow empty strings if they're required
@sbalaji1192

Comment (sbalaji1192) [reply app/client/src/widgets/DropdownWidget/widget/index.tsx:281]: yeah. true

**Review by sbalaji1192 — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@Tooluloope Tested this PR and the functionality works as expected

**Pull request merged**